examples/nlp/gpt/conf/gpt_self_rewarding.yaml

```diff
@@ -0,0 +1,281 @@
+      # Reloading the engine incurs a constant time overhead
+      unload_engine_train: False
+
+  # do not change these
+  logger: False # logger provided by exp_manager
+  enable_checkpointing: False
+  use_distributed_sampler: False
+  max_time: null
+  max_epochs: ${.self_rewarding.max_epochs}
+  max_steps: ${.self_rewarding.max_steps}
+
+exp_manager:
+  explicit_log_dir: /results
+  exp_dir: null
+  name: megatron_gpt
+  max_time_per_run: ${trainer.max_time}
+  create_wandb_logger: False
+  wandb_logger_kwargs:
+    project: nemo_aligner_dpo
+    name: rlhf_gpt3_dpo
+  resume_from_checkpoint: null # The path to a checkpoint file to continue the training, restores the whole state including the epoch, step, LR schedulers, apex, etc.
+  resume_if_exists: True
+  resume_ignore_no_checkpoint: True
+  create_checkpoint_callback: True
+  checkpoint_callback_params:
+    monitor: val_loss
+    save_top_k: 3
+    mode: min
+    always_save_nemo: False # saves nemo file during validation, not implemented for model parallel
+    save_nemo_on_train_end: True # not recommended when training large models on clusters with short time limits
+    filename: 'megatron_gpt--{${.monitor}:.3f}-{step}-{consumed_samples}-{iteration}-{epoch}'
+    model_parallel_size: ${multiply:${model.tensor_model_parallel_size}, ${model.pipeline_model_parallel_size}}
+
+pretrained_checkpoint:
+  restore_from_path: null
+
+model:  
+  mcore_gpt: True
+  micro_batch_size: 1
+  global_batch_size: 64
+  megatron_amp_O2: True
+  dist_ckpt_format: zarr
+
+  spin:
+    log_prob_forward_micro_batch_size: ${model.micro_batch_size}
+    rollout_micro_batch_size: 32
+    ref_policy_kl_penalty: 0.1   # Can also be a list of elements == max_iterations, where each element will be the KL penalty used for that iteration
+    offload_adam_states: True
+    num_responses_to_gen: 4
+    num_evals_to_average: 3
+    length_control: 0.0
+    use_meta_judge: False
+    meta_judge_pcnt: -1
+    first_iteration_sft: False
+    preference_average_log_probs: False # whether normalizing log probs according to the sequence length in preference_loss
+    sft_average_log_probs: ${.preference_average_log_probs} # whether normalizing log probs according to the sequence length in sft_loss
+    gt_reward_scale: 1. # the scale of the rewards in RPO
+    preference_loss: dpo # the preference loss, we support dpo, ipo, rpo_sq, rpo_bwd_kl, rpo_fwd_kl
+    preference_loss_weight: 1 # the coefficient of the preference loss
+    sft_loss_weight: 0 # the coefficient of the SFT loss
+    judge_score_low: 0  # the lowest score which the LLM-as-a-judge can aware, as per your llm_judge_prompt
+    judge_score_high: 5  # the highest score which the LLM-as-a-judge can aware, as per your llm_judge_prompt
+    meta_max_relative_pcnt: 0.4
+
+    judge_reward_regex: "(?i)(?:Score|Points): {{ reward }}"
+    meta_judge_reward_regex: "(?i)Winner: (?:Judgement|Judgment) {{ reward }}"
+
+    llm_judge_prompt: |
```

Comment:
```suggestion
    # Current default judge prompt is copied from https://arxiv.org/abs/2401.10020
    # Note that its template (the "<extra_id_..." lines) may need to be customized to your specific model
    llm_judge_prompt: |
```

Reply:
Accepted

Reply:
Actually we now need to remove "Note that its template (the "<extra_id_..." lines) may need to be customized to your specific model" (in both places)

Reply:
Done